sourcegraph/sourcegraph.proto

```diff
@@ -781,6 +786,20 @@ message ChangesetUpdateOp {
 	UserSpec Author = 8 [(gogoproto.nullable) = false];
 }
 
+message ChangesetUpdateAffectedOp {
+	// Repo holds the RepoSpec which received a commit.
+	RepoSpec repo = 1 [(gogoproto.nullable) = false];
+
+	// Branch is the name of the branch which was pushed to.
+	string branch = 2;
+
+	// Last is the SHA1 of the previous commit on the branch.
+	string last = 3;
+
+	// Commit is the SHA1 of the newly pushed commit on the branch.
+	string commit = 4;
```

Comment:
Good point. The last commit is necessary when the push operation contains multiple commits, so `Last` is not always the parent of `Commit`. These commit ids are provided in the [githttp.Event](https://sourcegraph.com/github.com/AaronO/go-git-http@master/.GoPackage/github.com/AaronO/go-git-http/.def/Event) emitted by push operations.

I updated the docstrings to reflect that a push operation invoking `UpdateAffected` may consist of multiple commits.